src/releasenotes_check_workflow/releasenotes_check.py

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+
+
+class ReleaseNotesCheck:
```

Comment:
While you've moved the check code into a class, you're not taking advantage of OOO. The implementation still has `if component == OpenSearch`, etc.

Refactor:

1. Create a base class called `ReleaseNotes` with an instance method `check`. The initializer takes a component.
2. Add a property called `ReleaseNotes#filename` that defaults to `.release-notes-{buildversion}.md`.
3. Add a method `ReleaseNotes#exists` that checks whether the file exists, don't rely on an exception.
4. Subclass `ReleaseNotes` into `ReleaseNotesOpenSearch` and `ReleaseNotesOpenSearchDashboards` that overrde `filename` to return a different name for those two.